lib/features/public_asset/domain/usecase/add_identity_to_public_assets_interactor.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:core/presentation/state/failure.dart';
+import 'package:core/presentation/state/success.dart';
+import 'package:dartz/dartz.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/session/session.dart';
+import 'package:jmap_dart_client/jmap/identities/identity.dart';
+import 'package:jmap_dart_client/jmap/mail/extensions/public_asset/public_asset.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/repository/public_asset_repository.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/state/add_identity_to_public_assets_state.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/state/remove_identity_from_public_assets_state.dart';
+import 'package:tmail_ui_user/features/public_asset/presentation/public_asset_controller.dart';
+
+class AddIdentityToPublicAssetsInteractor {
+  AddIdentityToPublicAssetsInteractor(this._publicAssetRepository);
+
+  final PublicAssetRepository _publicAssetRepository;
+
+  Stream<Either<Failure, Success>> execute(
+    Session session,
+    AccountId accountId,
+    {
+      required IdentityId identityId,
+      required List<PublicAssetId> publicAssetIds
+    }
+  ) async* {
+    try {
+      yield Right(AddingIdentityToPublicAssetsState());
+      final publicAssets = await _publicAssetRepository.getPublicAssetsFromIds(
+        session,
+        accountId,
+        publicAssetIds: publicAssetIds
+      );
+      if (publicAssets.isEmpty) {
+        yield Left(NotFoundAnyPublicAssetsFailureState());
+      } else {
+        final publicAssetsWithCurrentIdentity = publicAssets
+          .map((publicAsset) => PublicAsset(
+            id: publicAsset.id,
+            publicURI: publicAsset.publicURI,
+            size: publicAsset.size,
+            contentType: publicAsset.contentType,
+            blobId: publicAsset.blobId,
+            identityIds: publicAsset.identityIds?..addAll({identityId: true}),
```

Comment:
should create extension for `PublicAsset `

Reply:
Done